linera-execution/src/runtime.rs

```diff
@@ -933,54 +925,52 @@ impl<UserInstance> BaseRuntime for SyncRuntimeInternal<UserInstance> {
         content_type: String,
         payload: Vec<u8>,
     ) -> Result<Vec<u8>, ExecutionError> {
-        if let OracleResponses::Replay(responses) = &mut self.oracle_responses {
-            return match responses.next() {
-                Some(OracleResponse::Post(bytes)) => Ok(bytes),
-                Some(_) => Err(ExecutionError::OracleResponseMismatch),
-                None => Err(ExecutionError::MissingOracleResponse),
-            };
-        }
-        let url = url.to_string();
-        let bytes = self
-            .execution_state_sender
-            .send_request(|callback| ExecutionRequest::HttpPost {
-                url,
-                content_type,
-                payload,
-                callback,
-            })?
-            .recv_response()?;
-        if let OracleResponses::Record(responses) = &mut self.oracle_responses {
-            responses.push(OracleResponse::Post(bytes.clone()));
-        }
+        let bytes = if let Some(responses) = &mut self.replaying_oracle_responses {
+            match responses.next() {
+                Some(OracleResponse::Post(bytes)) => bytes,
+                Some(_) => return Err(ExecutionError::OracleResponseMismatch),
+                None => return Err(ExecutionError::MissingOracleResponse),
+            }
+        } else {
+            let url = url.to_string();
+            self.execution_state_sender
+                .send_request(|callback| ExecutionRequest::HttpPost {
+                    url,
+                    content_type,
+                    payload,
+                    callback,
+                })?
+                .recv_response()?
+        };
+        self.recorded_oracle_responses
+            .push(OracleResponse::Post(bytes.clone()));
```

Comment:
ok

Reply:
I'm reading that in a tone of voice that suggests you take objection…? :sweat_smile:

Reply:
not at all! Just signaling that I saw the change